bad_connector_woocommerce/models/product_template/common.py

```diff
@@ -0,0 +1,145 @@
+import logging
+
+from odoo import api, fields, models
+
+from odoo.addons.component.core import Component
+from odoo.addons.component_event import skip_if
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    woo_bind_ids = fields.One2many(
+        comodel_name="woo.product.template",
+        inverse_name="odoo_id",
+        string="WooCommerce Bindings",
+        copy=False,
+    )
+
+    variant_different = fields.Boolean()
+    stock_manage_template = fields.Boolean(
+        compute="_compute_stock_manage_template", store=True
+    )
+    backend_stock_manage = fields.Boolean(
+        compute="_compute_backend_stock_manage", store=True
+    )
+
+    @api.depends(
+        "woo_bind_ids",
+        "woo_bind_ids.stock_management",
+    )
+    def _compute_stock_manage_template(self):
+        """Compute the stock management status for each WooCommerce Product."""
+        for template in self:
+            template.stock_manage_template = any(
+                template.woo_bind_ids.mapped("stock_management")
+            )
+
+    @api.depends(
+        "woo_bind_ids",
+        "woo_bind_ids.backend_id.update_stock_inventory",
+    )
+    def _compute_backend_stock_manage(self):
+        for template in self:
+            template.backend_stock_manage = (
+                self.woo_bind_ids.backend_id.update_stock_inventory
+            )
+
+    def update_stock_qty(self):
```

Comment:
the same method is there in product.product level so it is better to put this common method in one place and use it wherever required

Reply:
This method is for button action which is located at product.template level, So for that reason we have to define this method here.